service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/AzureVmHelper.java

```diff
@@ -60,18 +60,7 @@ public static StepResult deleteNetworkInterface(
           .networkInterfaces()
           .deleteByResourceGroup(azureCloudContext.getAzureResourceGroupId(), networkInterfaceName);
     } catch (ManagementException e) {
-      // Stairway steps may run multiple times, so we may already have deleted this resource.
-      // For delete steps, this exception handling exists in DeleteAzureControlledResourceStep
-      // But this is also used in the undo part of CreateAzureNetworkInterfaceStep, so it's being
-      // left intact
       if (AzureManagementExceptionUtils.isExceptionCode(
-          e, AzureManagementExceptionUtils.RESOURCE_NOT_FOUND)) {
```

Comment:
That is correct.
The clause for `AzureManagementExceptionUtils.NIC_RESERVED_FOR_ANOTHER_VM` remains here, because it's specific to this call.

I'd also prefer that these helpers _not_ return a `StepResult`, and just care about the operation, not stairway - but refactors have to stop _somewhere_.